pkg/image/association.go

```diff
@@ -35,8 +35,18 @@ func (e *ErrInvalidComponent) Error() string {
 	return fmt.Sprintf("image %q has invalid component %q", e.image, e.tag)
 }
 
-// Associations is a set of image Associations.
-type Associations map[string]Association
+// AssociationSet is a mapped for Association
+// searching
+type AssociationSet map[*Association]struct{}
```

Comment:
Using a pointer as a map key is dangerous. I would use the association's ID value instead.

Reply:
The intent here was trying to map the associations struct to the images. Do you think it would be better to just use a a map with a slice as a values so we can have multiple values per key or would it be better to move the Association to the values in the AssociationSet and use the ID as the key?

Reply:
Resolving. Pointer as map key removed.